### ISSUE TITLE
Register mail layouts and mail partials in a plugin

### ISSUE THREAD

**Issue by TimFoerster** [octobercms/october#3820]
##### Expected behavior
A plugin should be able to publish its own mail layouts and partials, like the `System\Classes\PluginBase::registerMailTemplates` method. 

##### Actual behavior
There is no easy way to integrate layouts or partials of a plugin

##### October build
current

**Comment by LukeTowers** [comment]
@TimFoerster feel free to submit a PR to introduce that feature :) Will need an accompanying docs entry

**Comment by abnmt** [comment]
I register layouts and partials like this:

``` php
use System\Classes\MailManager;

class Plugin extends PluginBase
{
    public function register()
    {
        MailManager::instance()->registerCallback(function ($manager) {
            $manager->registerMailLayouts([
                'l-welcome' => 'akme.plugin::mail.layout-welcome',
            ]);
            $manager->registerMailPartials([
                'p-head'    => 'akme.plugin::mail.partial-head',
                'p-header'  => 'akme.plugin::mail.partial-header',
                'p-body'    => 'akme.plugin::mail.partial-body',
                'p-footer'  => 'akme.plugin::mail.partial-footer',
                'p-picture' => 'akme.plugin::mail.partial-picture',
            ]);
        });
    }
}

```

**Comment by TimFoerster** [comment]
implemented by #3850
